src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -46,89 +48,167 @@ class MatchSamplingRate(Transform):
+        transformed_data = pl.DataFrame()
+        for i in range(len(data.collect().rows())):
+            transformed_data_slice = self._transform_row(
+                data.collect().slice(i, 1),
+            )
+            transformed_data = transformed_data.vstack(transformed_data_slice)
+        return transformed_data.lazy()
+
+    def _transform_row(self, data: pl.DataFrame) -> pl.DataFrame:
+        non_timeseries_features = data.select(
+            pl.exclude(
+                *self.feature_interpolation_map.keys(),
+                self.reference_feature_name,
+            ),
+        )
+        debug_msg = (
+            "Interpolating timeseries features: "
+            f"{self.feature_interpolation_map.keys()} using the timestamps of "
+            f"the and reference feature {self.reference_feature_name}"
+        )
+        logger.debug(debug_msg)
+
+        if self.reference_feature_name not in data.columns:
+            msg = f"'{self.reference_feature_name}' not in the DataFrame."
+            raise FeatureNotFoundError(msg)
+
+        features = list(self.feature_interpolation_map.keys())
+
+        reference_feature = (
+            data.select(pl.col(self.reference_feature_name).explode())
+            .unnest(cs.all())
+            .rename({"value": self.reference_feature_name + "_value"})
+        )
+
+        result = pl.concat(
+            [
+                interpolate_feature(
+                    feature,
+                    data,
+                    reference_feature,
+                    self.feature_interpolation_map[feature],
                 )
-            for (
-                feature,
-                timestamp,
-            ) in self.feature_columns_with_timestamps.items():
-                timestamps = df[timestamp][i].to_numpy()
-                feature_data = df[feature][i].to_numpy()
-                if np.issubdtype(timestamps.dtype, np.datetime64):
-                    timestamps = (
-                        timestamps.astype("datetime64[us]").astype(np.int64)
-                        / 1e6
-                    )
-                resampled_timeseries = interp(
-                    reference_timestamps,
-                    timestamps,
-                    feature_data,
-                ).tolist()
-                df = (
-                    df.lazy()
-                    .with_row_index()
-                    .with_columns(
-                        pl.when(pl.col("index") == i)
-                        .then(resampled_timeseries)
-                        .otherwise(pl.col(feature))
-                        .alias(feature),
-                    )
-                    .drop("index")
-                    .collect()
+                for feature in features
+            ],
+            how="horizontal",
+        )
+
+        return pl.concat(
+            [
+                data.select(self.reference_feature_name),
+                result,
+                data.select(non_timeseries_features),
```

Comment:
This already is a selected expression, isn't it?

```suggestion
                non_timeseries_features,
```

Reply:
done